src/index.d.ts

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2023 Inclusive Design Research Centre, OCAD University
+ * All rights reserved.
+ *
+ * Licensed under the New BSD license. You may not use this file except in
+ * compliance with this License.
+ *
+ * You may obtain a copy of the License at
+ * https://github.com/inclusive-design/adaptive-palette/blob/main/LICENSE
+ */
+
+export type BciAvIdType = number | (string|number)[];
+
+export type OptionsType = {
+    label: string,
+    columnStart: number,
+    columnSpan: number,
+    rowStart: number,
+    rowSpan: number,
+    bciAvId: BciAvIdType
+};
+
+export type JsonPaletteType = {
+  name: string,
+  cells: {
+    [key: string]: {
+      type: string,
+      options: OptionsType
+    }
+  }
+};
+
+// Types associated with the `props` parameter of the component functions.
+
+export type PalettePropsType = {
+  json: JsonPaletteType
+};
+
+export type ActionBmwCodeCellPropsType = {
+  id: string,
+  options: OptionsType
+};
```

Comment:
These component specific type definitions can be defined in the component as a private type. What do you think?

Reply:
I just searched the code base, and you are correct.  The only place that  `ActionBmwCodeCellPropsType` is used is inside `ActionBmwCodeCell.ts`.  I'll move that one out of the shared index file and make it local.

Thanks for noticing that!

Reply:
Same with `PalettePropsType`. It's only used in `Palette.ts`.

Reply:
Done.  I also found that I missed a `BlissSymbolProps` type in terms of the new naming convention.  I fixed that too.